### PR TITLE
Vector comparison impr

### PR DESCRIPTION
**Pull request by pointhi**
Implement various operators for integer vectors and normal integer variables of type ```i1```, ```i8```, ```i16``` and ```i64```:

### CODE REVIEW

File: projects/com.oracle.truffle.llvm.nodes/src/com/oracle/truffle/llvm/nodes/op/arith/integer/LLVMI1ArithmeticNode.java [hunk 1]
```diff
@@ -44,4 +45,67 @@ protected boolean executeI1(boolean left, boolean right) {
...
+            return left ^ right;
+        }
+    }
+
+    public abstract static class LLVMI1DivNode extends LLVMI1ArithmeticNode {
+
+        @Specialization
+        protected boolean div(boolean left, boolean right) {
+            if (!right) {
+                CompilerAsserts.neverPartOfCompilation();
```
Comment (grimmerm) [inline projects/com.oracle.truffle.llvm.nodes/src/com/oracle/truffle/llvm/nodes/op/arith/integer/LLVMI1ArithmeticNode.java]: That should be CompilerDirectives.transferToInterpreter

File: projects/com.oracle.truffle.llvm.nodes/src/com/oracle/truffle/llvm/nodes/op/arith/integer/LLVMI1ArithmeticNode.java [hunk 2]
```diff
@@ -44,4 +45,67 @@ protected boolean executeI1(boolean left, boolean right) {
...
+            return left;
+        }
+    }
+
+    public abstract static class LLVMI1UDivNode extends LLVMI1ArithmeticNode {
+
+        @Specialization
+        protected boolean udiv(boolean left, boolean right) {
+            if (!right) {
+                CompilerAsserts.neverPartOfCompilation();
```
Comment (grimmerm) [inline projects/com.oracle.truffle.llvm.nodes/src/com/oracle/truffle/llvm/nodes/op/arith/integer/LLVMI1ArithmeticNode.java]: same here

File: projects/com.oracle.truffle.llvm.nodes/src/com/oracle/truffle/llvm/nodes/op/arith/vector/LLVMI1VectorArithmeticNode.java [hunk 3]
```diff
@@ -38,10 +38,59 @@
 @NodeChildren({@NodeChild("leftNode"), @NodeChild("rightNode")})
 public abstract class LLVMI1VectorArithmeticNode extends LLVMExpressionNode {
 
-    public abstract static class LLVMI1VectorAddNode extends LLVMI1VectorArithmeticNode {
+    public abstract static class LLVMI1VectorAddNode extends LLVMI8VectorArithmeticNode {
```
Comment (grimmerm) [inline projects/com.oracle.truffle.llvm.nodes/src/com/oracle/truffle/llvm/nodes/op/arith/vector/LLVMI1VectorArithmeticNode.java]: That does not look right.
how can I1 extend I8?

File: projects/com.oracle.truffle.llvm.runtime/src/com/oracle/truffle/llvm/runtime/vector/LLVMI1Vector.java [hunk 4]
```diff
@@ -88,6 +89,85 @@ private static LLVMI1Vector doOperation(LLVMI1Vector lhs, LLVMI1Vector rhs, Oper
...
+            }
+        });
+    }
+
+    public LLVMI1Vector div(LLVMI1Vector rightValue) {
+        return doOperation(this, rightValue, new Operation() {
+            @Override
+            public boolean eval(boolean a, boolean b) {
+                if (!b) {
+                    CompilerAsserts.neverPartOfCompilation();
```
Comment (grimmerm) [inline projects/com.oracle.truffle.llvm.runtime/src/com/oracle/truffle/llvm/runtime/vector/LLVMI1Vector.java]: transferToInterpreter

**Comment by pointhi** [comment]
Thanks @grimmerm for outlining some mistakes on my side.

I fixed them in https://github.com/pointhi/sulong/commit/6ad153c3d9f611392085687783fa3cdeb91e152d, and squashed all commits together

**Pull request closed without merging**